form.js

```diff
@@ -22,137 +22,115 @@ define('form', ['doc'], function($) {
-				valid = false;
+			if (isNaN(numericFieldValue) || condition(numericFieldValue, numericTypeValue)) {
+				return messageWithArgs(validationMessages[type], typeValue);
 			}
-		});
+		}
+	};
 
-		form.find('[pattern]').each(function(element){
-			var pattern = new RegExp('^' + element.getAttribute('pattern') + '$');
+	var getMinValidate = function($field) {
+		if (!$field.attr('min')) {
+			return false;
+		}
 
-			if ($(element).val() && !pattern.test($(element).val())) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.pattern);
-				valid = false;
-			}
+		return getMaxOrMinMessage($field, function(numericFieldValue, numericTypeValue) {
+			return numericFieldValue < numericTypeValue;
 		});
+	};
 
-		form.find('[type="email"]').each(function(element){
-			var pattern = new RegExp(/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/);
-
-			if (($(element).val()) && !pattern.test($(element).val())) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.email);
-				valid = false;
-			}
-		});
+	var getMaxValidate = function($field) {
+		if (!$field.attr('max')) {
+			return false;
+		}
 
-		form.find('[type="url"]').each(function(element){
-			if ($(element).val() && !element.validity.valid) {
-				var parent = element.parentElement;
-				appendMessage($(parent), validationMessages.url);
-				valid = false;
-			}
+		return getMaxOrMinMessage($field, function(numericFieldValue, numericTypeValue) {
+			return numericFieldValue > numericTypeValue;
 		});
+	};
 
-		if (!valid){
-			form.addClass('has-errors');
+	var getMaxLengthValidate = function($field) {
+		var length = $field.attr('maxlength'),
+			fieldValue = $field.val();
 
-			var visibleErrorFields = form.find('.error input, .error textarea').filter(function(field) {
-				return $(field).attr('type') !== 'hidden';
-			});
-
-			addEventToClearErrorMessages(visibleErrorFields);
-			this.focus(visibleErrorFields.first());
+		if (fieldValue && $field.attr('maxlength') !== null) {
+			if (fieldValue.length > length) {
+				return messageWithArgs(validationMessages.maxlength, length);
+			}
 		}
+	};
 
-		return valid;
+	var getRequiredValidate = function($field) {
+		if (!$field.val() && $field.attr('required') !== null) {
+			return validationMessages.required;
+		}
 	};
 
-	var addEventToClearErrorMessages = function(visibleErrorFields) {
-		visibleErrorFields.each(function(element){
-			$(element).on('input', function(event){
-				var target = event.target ? $(event.target) : $(window.event.srcElement),
-					parent = target.parent();
-				target.off('input', 'validationOff');
-				parent.removeClass('validation');
-				parent.removeClass('error');
-				parent.find('.validation-message').removeItem();
-			}, 'validationOff');
+	var isValid = function(form) {
```

Comment:
I think this parameter should be named `$form` and not `form` because it is a doc object

@tisvasconcelos